File: themeablecomponents/src/main/java/com/dertyp7214/themeablecomponents/colorpicker/ColorSeekBar.java
package com.dertyp7214.themeablecomponents.colorpicker;

import android.content.Context;
import android.content.res.ColorStateList;
import android.graphics.Color;
import android.util.AttributeSet;

import androidx.annotation.ColorInt;
import androidx.appcompat.widget.AppCompatSeekBar;

/**
 * Created by lengw on 20.09.2017.
 */

public class ColorSeekBar extends AppCompatSeekBar {

    public ColorSeekBar(Context context, AttributeSet attrs) {
        super(context, attrs);

        ColorStateList black = new ColorStateList(new int[][]{{}}, new int[]{Color.BLACK});

        setThumbTintList(black);
        setProgressTintList(black);
    }

    public void setColor(@ColorInt int c){
        ColorStateList color = new ColorStateList(new int[][]{{}}, new int[]{c});

        setThumbTintList(color);
        setProgressTintList(color);
    }
}

File: themeablecomponents/src/main/java/com/dertyp7214/themeablecomponents/colorpicker/ColorSeekBar.kt
package com.dertyp7214.themeablecomponents.colorpicker

import android.content.Context
import android.content.res.ColorStateList
import android.graphics.Color
import android.util.AttributeSet

import androidx.annotation.ColorInt
import androidx.appcompat.widget.AppCompatSeekBar

/**
 * Created by lengw on 20.09.2017.
 */

class ColorSeekBar(context: Context, attrs: AttributeSet) : AppCompatSeekBar(context, attrs) {

    init {
        val black = ColorStateList(arrayOf(intArrayOf()), intArrayOf(Color.BLACK))

        thumbTintList = black
        progressTintList = black
    }

    fun setColor(@ColorInt c: Int) {
        val color = ColorStateList(arrayOf(intArrayOf()), intArrayOf(c))

        thumbTintList = color
        progressTintList = color
    }
}
